src/main.rs

```diff
@@ -244,6 +247,20 @@ fn inner_main() -> Result<ExitCode> {
         .iter()
         .map(|path| FilePattern::from_user(path, &project_root))
         .collect();
+    let per_file_ignores: Vec<PerFileIgnore> = {
+        let (rv, rv_err): (Vec<_>, Vec<_>) = cli
+            .per_file_ignores
+            .iter()
+            .map(|pair| PerFileIgnore::from_user(pair, &project_root))
+            .partition(Result::is_ok);
+        if !rv_err.is_empty() {
+            rv_err.into_iter().for_each(|err| {
+                println!("{}", err.unwrap_err());
+            });
```

Comment:
I think we should just error hard if they provide a failing pattern or non-existent error code here. That's what we do with ignores and selects. Do you mind rewriting to fail-on-error?

Reply:
There now exists a type `crate::pyproject::StrCheckCodePair` with FromStr and Deserialize implemented such that errors can be caught in cli parsing and pyproject deserializing